src/status_im/chat/views/geolocation/views.cljs

```diff
@@ -0,0 +1,173 @@
+(ns status-im.chat.views.geolocation.views
+  (:require-macros [status-im.utils.views :refer [defview]]
+                   [reagent.ratom :refer [reaction]])
+  (:require [status-im.components.react :refer [view image text touchable-highlight]]
+            [reagent.core :as r]
+            [goog.string :as gstr]
+            [status-im.utils.utils :refer [http-get]]
+            [status-im.utils.types :refer [json->clj]]
+            [status-im.chat.views.geolocation.styles :as st]
+            [status-im.components.mapbox :refer [mapview]]
+            [re-frame.core :refer [dispatch subscribe]]
+            [status-im.i18n :refer [label]]
+            [status-im.components.react :as components]))
+
+(def mapbox-api "https://api.mapbox.com/geocoding/v5/mapbox.places/")
+(def access-token "pk.eyJ1Ijoic3RhdHVzaW0iLCJhIjoiY2oydmtnZjRrMDA3czMzcW9kemR4N2lxayJ9.Rz8L6xdHBjfO8cR3CDf3Cw")
+
+(defn get-places [coords cur-loc-geocoded & [poi?]]
+  (let [{:keys [latitude longitude]} coords]
+    (when (and latitude longitude)
+      (http-get (str mapbox-api longitude "," latitude
+                     ".json?" (when poi? "types=poi&") "access_token=" access-token)
+                #(reset! cur-loc-geocoded (json->clj %))
+                #(reset! cur-loc-geocoded nil))
+      true)))
+
+(defn place-item [{:keys [title address pin-style] [latitude longitude] :center}]
+  [touchable-highlight {:on-press #(do
+                                     (dispatch [:set-command-argument [0
+                                                                       (str (or address title)
+                                                                            "&amp;" latitude
+                                                                            "&amp;" longitude)
+                                                                       false]])
+                                     (dispatch [:send-seq-argument]))}
```

Comment:
not really a fan of long lines, and looks like this one can be shortened. but it's up to you.